### PR TITLE
docs: add openapi yaml file for api docs

### PR DESCRIPTION
**Pull request by dohaki**
This PR adds the OpenAPI yaml file for our API docs. We also updated the docs according to the changes made here https://github.com/across-protocol/frontend-v3/pull/1013. I opted to open a separate PR to not further increase the diff.

### CODE REVIEW

File: api-docs-openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+  version: 0.1.0
+servers:
+  - url: https://app.across.to/api
+paths:
+  /suggested-fees:
+    get:
+      summary: Retrieve suggested fee quote for a deposit.
+      description: >
+        Returns suggested fees based on the `token` OR
+        `inputToken`+`outputToken`,  `destinationChainId`, and `amount`.
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:14]: isn't `originChainId` required too?

File: api-docs-openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+            Note that this amount is in the native decimals of the token. So,
+            for WETH, this would be the amount of human-readable WETH multiplied
+            by 1e18. For USDC, you would multiply the number of human-readable
+            USDC by 1e6.
+
+
+            _Example:_ 100000000000
+          schema:
+            type: integer
+        - name: originChainId
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:112]: I thought we wanted to make this required at some point?

File: api-docs-openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+            higher precedence.**
+
+
+            _Example:_ 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
+        - name: inputToken
+          in: query
+          schema:
+            type: string
+          description: >
+            Address of token to bridge on the _origin_ chain. Must be used
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:261]: Is `originChain` implied/should we require it?

File: api-docs-openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+      responses:
+        "200":
+          description: Transfer limits
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TransferLimits"
+              example:
+                {
+                  "minDeposit": "7799819",
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:333]: I think it might be useful to define what these limits translate to in terms of ETA of fill

Comment (nicholaspai) [reply api-docs-openapi.yaml:333]: ah nvm i see you do this

File: api-docs-openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+            The maximum deposit size in the tokens' units. Note: The formatting
+            of this number is the same as minDeposit.
+          example: 22287428516241
+        maxDepositInstant:
+          type: string
+          description: >
+            The max deposit size that can be relayed "instantly" on the
+            destination chain. Instantly means that there is relayer capital
+            readily available and that a relayer is expected to relay within 1-4
+            minutes of the deposit.
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:551]: this can be less than 1 minute right?

File: api-docs-openapi.yaml [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+          description: >
+            The max deposit size that can be relayed with a "short delay" on the
+            destination chain. This means that there is relayer capital
+            available on mainnet and that a relayer will immediately begin
+            moving that capital over the canonical bridge to relay the deposit.
+            Depending on the chain, the time for this can vary. Polygon is the
+            worst case where it can take between 20 and 35 minutes for the
+            relayer to receive the funds and relay. Arbitrum is much faster,
+            with a range between 5 and 15 minutes. Note: if the transfer size is
+            greater than this, the estimate should be between 2-4 hours for a
```
Comment (nicholaspai) [inline api-docs-openapi.yaml:564]: This will change with CEX rebalancing FYI

File: api-docs-openapi.yaml (hunk 2 above)
Comment (dohaki) [reply api-docs-openapi.yaml:112]: We don't enforce it yet but I can make it required in the docs for now 🤔

File: api-docs-openapi.yaml (hunk 5 above)
Comment (dohaki) [reply api-docs-openapi.yaml:551]: yes will change

File: api-docs-openapi.yaml (hunk 1 above)
Comment (dohaki) [reply api-docs-openapi.yaml:14]: It is not currently. But I can state it as `required` in the docs before we enforce it eventually.

File: api-docs-openapi.yaml (hunk 3 above)
Comment (dohaki) [reply api-docs-openapi.yaml:261]: I think it makes sense yea. At least it doesn't hurt to state it as required in the docs.

File: api-docs-openapi.yaml [hunk 7]
```diff
@@ -0,0 +1,571 @@
...
+
+
+            **Also note that this parameter will eventually replace the
+            parameter `token` above. For now, you need to either specify `token`
+            OR `inputToken` +`outputToken` with the parameter `token` having a
+            higher precedence.**
+
+
+            _Example:_ 0x4200000000000000000000000000000000000006
+        - name: originChainId
```
Comment (james-a-morris) [inline api-docs-openapi.yaml:81]: Could we list a "currently supported" set of chain ids that would produce valid results?

File: api-docs-openapi.yaml [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+
+            Note that this amount is in the native decimals of the token. So,
+            for WETH, this would be the amount of human-readable WETH multiplied
+            by 1e18. For USDC, you would multiply the number of human-readable
+            USDC by 1e6.
+
+
+            _Example:_ 100000000000
+          schema:
+            type: integer
```
Comment (james-a-morris) [inline api-docs-openapi.yaml:116]: Is there a way to specify that this number should be positive?

File: api-docs-openapi.yaml [hunk 9]
```diff
@@ -0,0 +1,571 @@
...
+        - name: recipient
+          in: query
+          required: false
+          description: >
+            Recipient of the deposit. Can be an EOA or a contract. If this is an
+            EOA and message is defined, then the API will throw a 4xx error.
+
+
+            _Example_: 0xc186fa914353c44b2e33ebe05f21846f1048beda
```
Comment (james-a-morris) [inline api-docs-openapi.yaml:125]: For consistency - thoughts on making these addresses checksummed?

File: api-docs-openapi.yaml [hunk 10]
```diff
@@ -0,0 +1,571 @@
...
+          deprecated: true
+          schema:
+            type: string
+          description: >
+            Address of token to bridge. For native ETH, use the wrapped address,
+            i.e. WETH.
+
+
+            Note that the address provided must exist on the specified
+            `originChainId`.
```
Comment (james-a-morris) [inline api-docs-openapi.yaml:28]: We may want to include a small comment here about route collisions.

A possible example URL could be: https://app.across.to/api/limits?token=0xba100000625a3754423978a60c9317c58a424e3D&destinationChainId=1

Comment (dohaki) [reply api-docs-openapi.yaml:28]: We will make the parameter `originChainId` required in the docs at least. And enforce it soon in the API

File: api-docs-openapi.yaml (hunk 7 above)
Comment (dohaki) [reply api-docs-openapi.yaml:82]: https://github.com/across-protocol/frontend-v3/pull/1016/commits/d78cfc9109534bfb1e5a77ec0baa686a344fa9e1

File: api-docs-openapi.yaml (hunk 8 above)
Comment (dohaki) [reply api-docs-openapi.yaml:139]: https://github.com/across-protocol/frontend-v3/pull/1016/commits/d78cfc9109534bfb1e5a77ec0baa686a344fa9e1

File: api-docs-openapi.yaml (hunk 9 above)
Comment (dohaki) [reply api-docs-openapi.yaml]: https://github.com/across-protocol/frontend-v3/pull/1016/commits/d78cfc9109534bfb1e5a77ec0baa686a344fa9e1

**Review by james-a-morris — APPROVED** [review]

**Pull request merged**